coalib/results/result_actions/IgnoreResultAction.py

```diff
@@ -46,22 +47,32 @@ def apply(self, result, original_file_dict, file_diff_dict, language: str,
         """
         Add (I)gnore comment
         """
+        source_range = next(filter(lambda sr: exists(sr.file),
+                                   result.affected_code))
+        filename = source_range.file
+
+        original_line = original_file_dict[filename][
+                        source_range.start.line-1].rstrip()
 
-        ignore_comment = self.get_ignore_comment(result.origin, language)
+        ignore_comment = self.get_ignore_comment(result.origin, language,
+                                                 original_line)
 
         if not ignore_comment:
             return file_diff_dict
 
-        source_range = next(filter(lambda sr: exists(sr.file),
-                                   result.affected_code))
-        filename = source_range.file
-
         ignore_diff = Diff(original_file_dict[filename])
-        ignore_diff.change_line(
-            source_range.start.line,
-            original_file_dict[filename][source_range.start.line-1],
-            original_file_dict[filename][source_range.start.line-1].rstrip() +
-            '  ' + ignore_comment)
+
+        # Add the new ignore using 'and' with the previous ignore.
+        if ignore_comment == result.origin + ' and':
```

Comment:
Since both `and` and `comma` work fine I think we should prefer `comma` over `and`.